File: frontend/src/app/models/wallpaper.ts
export type Orientation = 'landscape' | 'portrait';

export interface DesktopPreset {
  id: string;
  label: string;
  width: number;
  height: number;
  aspect: string;
  orientation: Orientation;
  steps: number;
  guidance_scale: number;
}

export interface GenerateRequest {
  prompt: string;
  orientation: Orientation;
  preset: string;
  publish: boolean;
}

export interface GenerateResponse {
  id: string;
  prompt: string;
  width: number;
  height: number;
  orientation: Orientation;
  seed: number;
  steps: number;
  guidance_scale: number;
  model_id: string;
  image_url: string;
  mock: boolean;
  published: boolean;
  render_width?: number | null;
  render_height?: number | null;
}

export interface GalleryItem {
  id: string;
  prompt: string;
  width: number;
  height: number;
  orientation: Orientation;
  seed: number;
  steps: number;
  guidance_scale: number;
  model_id: string;
  image_url: string;
  published_at: string;
}

export interface HealthResponse {
  status: string;
  model_id: string;
  device: string;
  model_loaded: boolean;
  mock_generation: boolean;
  cuda_available: boolean;
  gpu_name?: string | null;
  vram_total_gb?: number | null;
}

export type ApplyTarget = 'background' | 'lockscreen';

File: mobile/lib/models/wallpaper.dart
class WallpaperPreset {
  WallpaperPreset({
    required this.id,
    required this.label,
    required this.width,
    required this.height,
    required this.aspect,
    required this.orientation,
    required this.steps,
    required this.guidanceScale,
  });

  final String id;
  final String label;
  final int width;
  final int height;
  final String aspect;
  final String orientation;
  final int steps;
  final double guidanceScale;

  factory WallpaperPreset.fromJson(Map<String, dynamic> json) {
    return WallpaperPreset(
      id: json['id'] as String,
      label: json['label'] as String,
      width: json['width'] as int,
      height: json['height'] as int,
      aspect: json['aspect'] as String,
      orientation: json['orientation'] as String,
      steps: json['steps'] as int,
      guidanceScale: (json['guidance_scale'] as num).toDouble(),
    );
  }
}

class GenerateResult {
  GenerateResult({
    required this.id,
    required this.prompt,
    required this.width,
    required this.height,
    required this.orientation,
    required this.seed,
    required this.steps,
    required this.guidanceScale,
    required this.modelId,
    required this.imageUrl,
    required this.mock,
    required this.published,
  });

  final String id;
  final String prompt;
  final int width;
  final int height;
  final String orientation;
  final int seed;
  final int steps;
  final double guidanceScale;
  final String modelId;
  final String imageUrl;
  final bool mock;
  final bool published;

  factory GenerateResult.fromJson(Map<String, dynamic> json) {
    return GenerateResult(
      id: json['id'] as String,
      prompt: json['prompt'] as String,
      width: json['width'] as int,
      height: json['height'] as int,
      orientation: json['orientation'] as String,
      seed: json['seed'] as int,
      steps: json['steps'] as int,
      guidanceScale: (json['guidance_scale'] as num).toDouble(),
      modelId: json['model_id'] as String,
      imageUrl: json['image_url'] as String,
      mock: json['mock'] as bool? ?? false,
      published: json['published'] as bool? ?? false,
    );
  }

  GenerateResult copyWith({bool? published}) {
    return GenerateResult(
      id: id,
      prompt: prompt,
      width: width,
      height: height,
      orientation: orientation,
      seed: seed,
      steps: steps,
      guidanceScale: guidanceScale,
      modelId: modelId,
      imageUrl: imageUrl,
      mock: mock,
      published: published ?? this.published,
    );
  }
}

class GalleryItem {
  GalleryItem({
    required this.id,
    required this.prompt,
    required this.width,
    required this.height,
    required this.orientation,
    required this.seed,
    required this.imageUrl,
    required this.publishedAt,
  });

  final String id;
  final String prompt;
  final int width;
  final int height;
  final String orientation;
  final int seed;
  final String imageUrl;
  final String publishedAt;

  factory GalleryItem.fromJson(Map<String, dynamic> json) {
    return GalleryItem(
      id: json['id'] as String,
      prompt: json['prompt'] as String,
      width: json['width'] as int,
      height: json['height'] as int,
      orientation: json['orientation'] as String,
      seed: json['seed'] as int,
      imageUrl: json['image_url'] as String,
      publishedAt: json['published_at'] as String,
    );
  }
}
